### PR TITLE
feat: add ECIES property value encryption

### PR DESCRIPTION
**Pull request by aleks-tpom6oh**
## What changed?

- **AvoEncryption.h/.m** — Standalone ECIES encryption class (P-256 + AES-256-GCM) with support for both compressed and uncompressed EC public keys
- **AvoNetworkCallsHandler** — Added `publicEncryptionKey` to init, `shouldEncrypt` gating (dev/staging only), `addEncryptedValues` recursive walker that encrypts primitive property values (skips lists, recurses objects), `jsonStringifyValue` helper, and `publicEncryptionKey` in base request body
- **AvoBatcher** — Threaded `eventProperties` parameter through `handleTrackSchema:` to enable encryption in the batched path
- **AvoInspector** — New init methods accepting `publicEncryptionKey` (backwards compatible), threaded `eventProperties` through all code paths (batched + validated)
- **Tests** — 29 new tests: 11 unit tests (round-trip encrypt/decrypt, format validation, compressed key support, edge cases) and 18 integration tests (encryption in dev/staging, no encryption in prod, nested objects, lists skipped, shouldEncrypt logic, jsonStringifyValue)
- **LocalConfig.h** — Gitignored local API key override mechanism with `__has_include` fallback

Mirrors [android-avo-inspector PR #38](https://github.com/avohq/android-avo-inspector/pull/38).

## How to test this PR:

- [ ] Run the test suite: `xcodebuild test` — all 153 tests pass (including 29 new encryption tests)
- [ ] Init `AvoInspector` with a P-256 public key in dev mode, track an event with properties, and verify `encryptedPropertyValue` appears in the network request body
- [ ] Verify no encryption occurs in prod mode even with a key provided
- [ ] Verify existing init methods without `publicEncryptionKey` still work (backwards compatible)

## Before merging
- [x] Security impact of change has been considered

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added support for encrypting property values using public encryption keys during initialization
  * Extended event tracking to accept and transmit event properties alongside schema data
  * Added new initialization options to configure public encryption keys and custom endpoints

* **Chores**
  * Updated configuration system to support local environment overrides

### CODE REVIEW

**Comment by aleks-tpom6oh** [comment]
@coderabbitai review

File: AvoInspector/Classes/AvoEncryption.m [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+
+#import "AvoEncryption.h"
+#import <Security/Security.h>
+#import <CommonCrypto/CommonCrypto.h>
+#import <CommonCrypto/CommonCryptor.h>
+
+// Forward-declare GCM functions that are in CommonCrypto but not always in public headers
+extern CCCryptorStatus CCCryptorGCMSetIV(CCCryptorRef cryptorRef, const void *iv, size_t ivLen);
+extern CCCryptorStatus CCCryptorGCMEncrypt(CCCryptorRef cryptorRef, const void *dataIn, size_t dataInLength, void *dataOut);
+extern CCCryptorStatus CCCryptorGCMFinalize(CCCryptorRef cryptorRef, void *tagOut, size_t tagLength);
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m]: Fixed in 024a039. Replaced deprecated multi-step CCCryptorGCMSetIV/CCCryptorGCMEncrypt/CCCryptorGCMFinalize with CCCryptorGCMOneshotEncrypt/CCCryptorGCMOneshotDecrypt — the modern oneshot API.

These are still forward-declared because all AES-GCM functions in CommonCrypto are SPI (not in public headers). Apple's public alternative, CryptoKit, is Swift-only and cannot be called from Objective-C without a bridging layer. This is the standard approach for ObjC AES-GCM. Added a detailed comment explaining the rationale.

File: AvoInspector/Classes/AvoEncryption.m [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+                memcpy(r, temp, 32);
+            }
+            uint8_t temp2[32];
+            [self bigMulMod:b b:b p:p result:temp2];
+            memcpy(b, temp2, 32);
+        }
+    }
+
+    memcpy(result, r, 32);
+}
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m:382]: Fixed in 024a039. Added a known secp256r1 test vector — the generator point G (X = 6B17D1F2..., Y = 4FE342E2...) — that validates the decompression is mathematically correct. The test encrypts with the compressed generator point; if decompression produces an incorrect Y, SecKeyCreateWithData rejects the key and the test fails.

Additionally, the round-trip compressed key test (which generates a random keypair, compresses it, encrypts, and decrypts successfully) already validates correctness end-to-end. Both tests pass.

File: AvoInspector/Classes/AvoEncryption.m [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+                   key:(NSData *)key
+                    iv:(NSData *)iv
+            ciphertext:(NSMutableData *)ciphertext
+               authTag:(NSMutableData *)authTag {
+    // Use step-by-step CCCryptorGCM API for broad iOS compatibility
+    CCCryptorRef cryptorRef = NULL;
+    CCCryptorStatus status;
+
+    status = CCCryptorCreateWithMode(kCCEncrypt,
+                                      11, // kCCModeGCM = 11
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m]: The `11` was `kCCModeGCM` — no longer needed. The fix in 024a039 replaced the entire multi-step GCM flow (CCCryptorCreateWithMode + SetIV + Encrypt + Finalize) with a single `CCCryptorGCMOneshotEncrypt` call that uses `kCCAlgorithmAES` directly. The magic number is gone.

File: AvoInspector/Classes/AvoEncryption.m [hunk 4]
```diff
@@ -0,0 +1,566 @@
...
+    // Encrypt
+    status = CCCryptorGCMEncrypt(cryptorRef, plaintext.bytes, plaintext.length, ciphertext.mutableBytes);
+    if (status != kCCSuccess) {
+        CCCryptorRelease(cryptorRef);
+        return NO;
+    }
+
+    // Finalize and get auth tag
+    size_t tagLength = kAuthTagLength;
+    status = CCCryptorGCMFinalize(cryptorRef, authTag.mutableBytes, tagLength);
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m]: Fixed in 024a039 — the oneshot API takes `tagLength` as a direct parameter (`kAuthTagLength`), so the `authTag.mutableBytes`/`authTag.length` inconsistency is no longer an issue.

File: Example/AvoStateOfTracking/AVOAppDelegate.m [hunk 5]
```diff
@@ -8,15 +8,24 @@
...
+#import "LocalConfig.h"
+#endif
+
+#ifndef AVO_INSPECTOR_API_KEY
+#define AVO_INSPECTOR_API_KEY @"A4lTbBQTGVyD1f66213X"
+#endif
+
 @implementation AVOAppDelegate
 
 AvoInspector * avoInspector;
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply Example/AvoStateOfTracking/AVOAppDelegate.m:22]: This is the example app (not the library itself) — it's only used for manual testing during development. The existing pattern with a file-scoped variable matches the original code structure and is sufficient for this purpose. Not changing.

File: Example/AvoStateOfTracking/AVOAppDelegate.m [hunk 6]
```diff
@@ -8,15 +8,24 @@
...
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
     // Override point for customization after application launch.
-    avoInspector = [[AvoInspector alloc] initWithApiKey:@"A4lTbBQTGVyD1f66213X" env: AvoInspectorEnvDev];
-    
+    avoInspector = [[AvoInspector alloc] initWithApiKey:AVO_INSPECTOR_API_KEY env: AvoInspectorEnvDev publicEncryptionKey: @"024ec9c17ea2fb3e727d2815941eeb7d7c6e551536c9e2dde37fbbf0ffb9850579"]; // example key, decrypt with 2f6f681eb9b9ce4d72c40fc3c860e1b484fc9001ceb8de162c0086c1c8c619cf
+
```
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply Example/AvoStateOfTracking/AVOAppDelegate.m:28]: This is a test-only P-256 key pair used in the example app for local development. The "private key" in the comment is the decryption counterpart to the public key, provided so developers can verify encrypted payloads during manual testing. It has no security value — it's not used in production, never protects real data, and the example app already ships with a hardcoded API key on the same line. Removing it would make the example app less useful for testing. Not changing.

File: AvoInspector/Classes/AvoEncryption.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleks-tpom6oh) [reply AvoInspector/Classes/AvoEncryption.m]: Thanks for the thorough follow-up. We've considered the CryptoKit wrapper approach but are keeping the current implementation for these reasons:

**App Store rejection risk is theoretical, not practical.** The App Store ingestion process flags usage of *private frameworks* (e.g., SpringBoard, GraphicsServices), not individual symbols within public frameworks. `CCCryptorGCMOneshotEncrypt` is an exported symbol in `libcommonCrypto.dylib`, which is part of the public CommonCrypto framework — the same framework we already link for SHA-256. These GCM functions have been stable exports since iOS 6 and are used by countless production iOS apps and libraries (JWT libraries, banking SDKs, etc.) without App Store issues.

**The "apps broke on iOS 15" concern doesn't apply here.** Documented SPI breakages involve private UI/system frameworks (UIKit internals, SpringBoard), not CommonCrypto. The GCM symbols have been in the stable ABI for ~14 years across every iOS release.

**Adding Swift has real costs for an ObjC CocoaPods library.** A CryptoKit wrapper would require: adding Swift runtime dependency for all consumers, module map changes, `s.swift_version` in podspec, and bumping the deployment target from iOS 12 to iOS 13 (breaking change). This is significant overhead for a function that works reliably today.

**If Apple ever removes these symbols** (extremely unlikely given 14 years of stability), we'll add the CryptoKit wrapper at that point — it's a straightforward change. But adding complexity preemptively for a risk that hasn't materialized in over a decade isn't warranted.
